src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java

```diff
@@ -303,7 +303,9 @@ public ListBoxModel doFillProfileItems(@RelativePath("..") @QueryParameter Strin
                 }
             }
 
-            profiles.get(0).selected = true;
+            if (profiles.size() == 1 && !"".equals(profiles.get(0).name)) {
```

Comment:
Yes, this code is just to auto-select first element of the combo if it only has 1 element (and it is not a default element without valid Id)